Library/Homebrew/formula.rb

```diff
@@ -1674,8 +1674,13 @@ def std_cmake_args(install_prefix: prefix, install_libdir: "lib", find_framework
            ldflags: T.nilable(T.any(String, T::Array[String]))).returns(T::Array[String])
   }
   def std_go_args(output: bin/name, ldflags: nil)
+    default_ldflags = "-s -w"
+
     args = ["-trimpath", "-o=#{output}"]
-    args += ["-ldflags=#{Array(ldflags).join(" ")}"] if ldflags
+
+    # Ensure there is no dups for `-s -w` in ldflags
+    combined_ldflags = ldflags&.include?(default_ldflags) ? [ldflags] : [default_ldflags, ldflags].compact
```

Comment:
`ldflags` could be either a string or an array of strings. I'm not sure this construction necessarily catches every possible inclusion of `-s` or `-w`. **I have not tested this**, but consider in formula:

```ruby
my_ldflags = %w[-s -w -X some.var=foo]
system "go", "build", *std_go_args(ldflags: my_ldflags)
```

Since the `%w` array splits `-s` and `-w` into separate elements, then `ldflags` does not include `-s -w` together in one string so these flags are repeated. This also doesn't catch the possibility of a single `-s` or `-w`, or in the opposite order `-w -s`.

Maybe these are all corner cases that we don't actually see, but could be something to consider.

Reply:
yeah, I can improve the syntax, but I have tested the dup `-s -w -s -w` works fine for the `go build`, not sure if it is worth the efforts.

Reply:
Please use `if`/`else` here, it's much easier to follow what's going on.